pkg/network/kubemacpool.go

```diff
@@ -0,0 +1,48 @@
+package network
+
+import (
+	"os"
+	"path/filepath"
+	"reflect"
+
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/render"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+)
+
+func changeSafeKubeMacPool(prev, next *opv1alpha1.NetworkAddonsConfigSpec) []error {
+	if prev.KubeMacPool != nil && !reflect.DeepEqual(prev.KubeMacPool, next.KubeMacPool) {
+		return []error{errors.Errorf("cannot modify KubeMacPool configuration once it is deployed")}
+	}
+	return nil
+}
+
+// renderLinuxBridge generates the manifests of Linux Bridge
+func renderKubeMacPool(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string) ([]*unstructured.Unstructured, error) {
+	if conf.KubeMacPool == nil {
+		return nil, nil
+	}
+
+	// render the manifests on disk
+	data := render.MakeRenderData()
+	data.Data["KubeMacPoolImage"] = os.Getenv("KUBEMACPOOL_IMAGE")
+	data.Data["ImagePullPolicy"] = os.Getenv("IMAGE_PULL_POLICY")
+
+	if conf.KubeMacPool.StartPoolRange == "" || conf.KubeMacPool.EndPoolRange == "" {
+		data.Data["ConfigurePoolRange"] = false
```

Comment:
this suggests, that if user does not pass MAC range, we don't create the configmap. However, it is not documented in the README.

Anyway, is there a good reason to deploy kubemacpool without the range configured? Maybe we should make it simpler and always create it. Either make the range mandatory or add defaulting.

Reply:
for now the kubemacpool doesn't generate so I remove it from this PR  and make it mandatory.
I will open a separate one after the kubemacpool support it.

I removed the omitempty from them